src/P/List.hs

```diff
@@ -45,3 +46,11 @@ lastMaybe = listToMaybe . reverse
 -- | count the number of elements satisfying a predicate in a list
 count :: (a -> Bool) -> [a] -> Int
 count predicate = length . filter predicate
+
+safeIndex :: [a] -> Int -> Maybe a
+safeIndex xs i
```

Comment:
@charleso I wrote it to have the same arg order as `!!` and used it as an infix operator.